### ISSUE TITLE
BUG: "Conversion from an array_api_strict array to a NumPy ndarray is not supported"

### ISSUE THREAD

**Issue by mdhaber** [scipy/scipy#21828]
### Describe your issue.

Essentially all array API tests (and some others) are failing. See e.g. gh-21823, gh-21831.

Crossref data-apis/array-api-strict#67.

### Reproducing Code Example

```python
-
```

### Error message

```shell
-
```

### SciPy/NumPy/Python version and system information

```shell
main
```

**Comment by lucascolley** [comment]
suggested way forward at https://github.com/data-apis/array-api-strict/pull/69#issuecomment-2464402872

**Comment by lucyleeow** [comment]
Since this removal of `__array__` was reverted, could this issue be considered out-dated?

**Comment by lucascolley** [comment]
Yes, we can close this 👍 I'll try to make some progress on https://github.com/scipy/scipy/pull/22049 this summer, which would help towards unblocking the removal of `__array__`.
